app/CodeFormatterApp.php

```diff
@@ -192,11 +192,11 @@ protected function putCodeInTempFile(string $file_key): string
      */
     protected function generateTempFileName(string $file_key)
     {
-        // generate sha256-hashed unique filename based on code content and key for specific part of post content
-        $filename = hash('sha256', $this->code_content . uniqid($file_key));
-        $file_extension = '.' . $this->code_language;
+        $filename = tempnam(__DIR__ . '/../storage/code', $file_key);
+        rename($filename, $filename .= '.' . $this->code_language);
```

Comment:
this is not guaranteed to be unique either, i know. this rename could collide with a existing file.